src/lib/radio/_radio-theme.scss

```diff
@@ -9,33 +9,57 @@
   $background: map-get($theme, background);
   $foreground: map-get($theme, foreground);
 
-  .mat-radio-outer-circle {
-    border-color: mat-color($foreground, secondary-text);
-  }
+    .mat-radio-button {
+        //default color (accent)
+        &:not(.mat-primary):not(.mat-accent):not(.mat-warn) {
+            @include radio-theme-color($accent, $foreground);
+        }
 
-  .mat-radio-checked .mat-radio-outer-circle {
-    border-color: mat-color($accent);
-  }
+        &.mat-primary {
+            @include radio-theme-color($primary, $foreground);
+        }
 
-  .mat-radio-disabled .mat-radio-outer-circle {
-    border-color: mat-color($foreground, disabled);
-  }
+        &.mat-accent {
+            @include radio-theme-color($accent, $foreground);
+        }
 
-  .mat-radio-inner-circle {
-    background-color: mat-color($accent);
-  }
+        &.mat-warn {
+            @include radio-theme-color($warn, $foreground);
+        }
+    }
+}
 
-  .mat-radio-ripple .mat-ripple-element {
-    background-color: mat-color($accent, 0.26);
-  }
+@mixin radio-theme-color($color, $foreground) {
```

Comment:
This mixin is going to generate a lot of css output; only styles that are directly affected by the palette swap should be defined within.